crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -1422,6 +1513,21 @@ pub(super) fn parse_location(location: &str, code: StatusCode) -> Result<Locatio
         .map_err(Into::into)
 }
 
+pub(super) fn require_allowed_location(
+    location: &Location,
+    storage_profile: &StorageProfile,
+) -> Result<()> {
+    if !storage_profile.is_allowed_location(location) {
+        return Err(ErrorModel::bad_request(
+            format!("Specified table location is not allowed: {location}"),
```

Comment:
add a small explanation what allowed locations are?

Reply:
I added a docstring to the `is_allowed_location` function.
I don't want to comment it in `table.rs`, because it might change in the future. That's what we have the function for. Currently its only sublocations of the one base location of a storage profile, eventually we might allow multiple locations per Storage Profile.